config/runtime/kunit.jinja2

```diff
@@ -142,6 +142,8 @@ cd {src_path}
     def _submit(self, result, node, api):
         # Ensure top-level name is kept the same
         result = result.copy()
+        # update node to newest
+        node = api.node.get(node['id'])
```

Comment:
These lines are not required as `api_helper.submit_results` already takes care of it.

Reply:
I had to remove it from submit_results, because artifacts get removed, for example from lava_callback, just noticed that yesterday.

As you can see in [lava-callback](https://github.com/kernelci/kernelci-pipeline/blob/e79e3743c2caeb106b99e7eaabd5904d0bca950f/src/lava_callback.py#L79) we are adding artifact to job_node, it is second argument to submit_results.
But in [submit_results](https://github.com/kernelci/kernelci-core/blob/b68f435edf231240e724d9c633ed787f4bb88e3d/kernelci/api/helper.py#L194) we are discarding everything and just using id to fetch fresh node data, which leads to complicated situation, that any added data to node will be discarded. And if we try to handle it in submit_results, it might be different in each case, which node and field have priority.
For example kunit job_node is generated directly during job creation, and doesn't contain jobid, but lava-callback have job_node retrieved "late", only when callback received and might contain job id, so it will be logical to handle this logic in each job differently, and make submit_results more simple, but flexible.

Reply:
https://staging.kernelci.org:9000/viewer?node_id=65cd5b8f76955310afcf8227
This is also example, where all works ok, because lava-callback naturally supply updated node data, and only kunit need refresh in the right place.

Reply:
At moment it is fixing bug, otherwise we will not have lava_log, so we have to come to some conclusion in reasonable time :) And i should fix this bug before deploying on production at monday.

Reply:
I will prefer right solution, this is why i am asking

Reply:
> Another way to go around it would be to fetch the node from the DB, then take the root node as passed to the function and merge them properly (with any existing field in the DB node overwritten by the new values passed in the parameter node). But this needs to be documented very clearly!

I completely agree with what Ricardo has pointed out here.

@nuclearcat If you want to make a quick fix without waiting for `submit_results` to get fixed, the simplest solution would be to update the node from lava callback before calling submit results. That would preserve data in `submit_results` method as well.

Reply:
But that also means removing node update from submit results?
Otherwise any new data i provide with node to submit_results is getting discarded.

Reply:
That still doesn't solve problem of extra API call at lava-callback and thats fundamental problem of submit_results.
In lava-callback node is _fresh_, because it does not have node, just id, when callback received.
In kunit (and similar runtimes that can submit job by themself) - they have copy of whole "stale" node created at moment of initial job/node creation that is guaranteed to not have job_id and other fields, this is why we need to fetch fresh copy of node.
I will add your solution atm into submit results just to fix bug, open issue and we will need to solve/discuss that properly.